core/src/calleeFunctions/index.ts

```diff
@@ -24,15 +27,44 @@ export const getCalleeData = async function (
     network: string,
     collateralType: string,
     marketId: string,
-    profitAddress: string
+    profitAddress: string,
+    pools?: Pool[]
 ): Promise<string> {
     const collateral = getCollateralConfigByType(collateralType);
     const marketData = collateral.exchanges[marketId];
     if (!marketData || !marketData.callee || !allCalleeFunctions[marketData.callee]) {
         throw new Error(`Unsupported collateral type "${collateralType}"`);
     }
-    return await allCalleeFunctions[marketData.callee].getCalleeData(network, collateral, marketId, profitAddress);
+    return await allCalleeFunctions[marketData.callee].getCalleeData(network, collateral, marketId, profitAddress, pools);
 };
+const getCalleeAutoRoute = async (
+    network: string,
+    collateral: CollateralConfig,
+    marketId: string,
+    amount: BigNumber = new BigNumber('1')
+): Promise<string[] | undefined> => {
+    const calleeConfig = collateral.exchanges[marketId];
+    if (!('automaticRouter' in calleeConfig)) {
+        return undefined;
+    }
+    return (await fetchAutoRouteInformation(network, collateral.symbol, amount.toFixed())).route || [];
+}
+
+const getMarketDataWithoutPrice = (calleeConfig: CalleeConfig, route: string[] | undefined, pools: Pool[] | undefined) => {
```

Comment:
- Not sure what this function is and what it is doing in `calleeFunction`
- You don't need to pass `route: string[] | undefined, pools: Pool[] | undefined` you only need to pass one thing. Route should not be used beyond the enrichment step. Making one or the other available and spreading `'automaticRouter' in calleeConfig` beyond enrichment is not desirable